Sources/SchafKit/Extensions/Collections/Array.swift

```diff
@@ -100,6 +100,50 @@ public extension Array {
         }
         return false
     }
+    
+    /// Get the max value of a field of the array element type.
+    ///
+    /// ```
+    /// ["foo", "bar", "toast"].max(by: \.count)
+    /// ```
+    ///
+    /// - Parameter f: A function/keypath returning the value to be compared.
+    func max<T: Comparable>(by f: (Element) -> T) -> T? {
+        self.map(f).max()
+    }
+    
+    /// Get the max value of a field of the array element type.
+    ///
+    /// ```
+    /// ["foo", "bar", "toast"].max(by: \.count)
+    /// ```
+    ///
+    /// - Parameter f: A function/keypath returning the optional value to be compared.
+    func max<T: Comparable>(by f: (Element) -> T?) -> T? {
+        self.compactMap(f).max()
+    }
+    
+    /// Get the min value of a field of the array element type.
+    ///
+    /// ```
+    /// ["foo", "bar", "toast"].min(by: \.count)
+    /// ```
+    ///
+    /// - Parameter f: A function/keypath returning the value to be compared.
+    func min<T: Comparable>(by f: (Element) -> T) -> T? {
```

Comment:
This function signature with `by` is also misleading, it should probably be 
```Swift
Array<T>.min(for: (Element) -> T) -> T?
```

If you wanna use `by` it would instead return an `Element` type.

Reply:
Not sure if `for` is the correct wording here either.
Should we go with `using mappingFunc` or `with mappingFunc`?

Reply:
Let's go with `of mappingFunc`.

Reply:
Done